src/legacy/ui/public/chrome/directives/header_global_nav/components/header.tsx

```diff
@@ -235,31 +238,27 @@ class HeaderUI extends Component<Props, State> {
     const leftNavControls = navControls.bySide[NavControlSide.Left];
     const rightNavControls = navControls.bySide[NavControlSide.Right];
 
-    let navLinksArray = navLinks.map(navLink =>
-      navLink.hidden || !uiCapabilities.navLinks[navLink.id]
-        ? null
-        : {
-            key: navLink.id,
-            label: navLink.title,
-            href: navLink.href,
-            iconType: navLink.euiIconType,
-            icon:
-              !navLink.euiIconType && navLink.icon ? (
-                <EuiImage
-                  size="s"
-                  alt=""
-                  aria-hidden={true}
-                  url={chrome.addBasePath(`/${navLink.icon}`)}
-                />
-              ) : (
-                undefined
-              ),
-            isActive: navLink.active,
-            'data-test-subj': 'navDrawerAppsMenuLink',
-          }
-    );
-    // filter out the null items
-    navLinksArray = navLinksArray.filter(item => item !== null);
+    const navLinksArray = navLinks
+      .filter(navLink => !navLink.hidden)
+      .map(navLink => ({
+        key: navLink.id,
+        label: navLink.title,
+        href: addBasePath(navLink.href),
+        iconType: navLink.euiIconType,
+        icon:
+          !navLink.euiIconType && navLink.icon ? (
```

Comment:
You can remove the explicit undefined with:

```js
icon: !navLink.euiIconType && navLink.icon && (
  <EuiImage
    size="s"
    alt=""
    aria-hidden={true}
    url={chrome.addBasePath(`/${navLink.icon}`)}
  />
),
```

Of course, that assumes that `icon` will be `undefined` and not some other falsy value, so if that's the case, please  leave this as is.